docs/event-modules.md

```diff
@@ -0,0 +1,96 @@
+## An tutorial/example of writing an event module for Warewulf
+
+### The module itself
+
+The below example is used to generate a node list file automatically within Warewulf, but can be co-opted to do anything you wish. The file is broken into functional components with descriptions of each component provided. To reassemble, simply copy and paste all of the code sections into a single file and put it into the Warewulf/Event/ perl library directory.
+
+#### Preamble and includes
+
+```
+# Copyright (c) 2001-2003 Gregory M. Kurtzer
+#
+# Copyright (c) 2003-2013, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory (subject to receipt of any
+# required approvals from the U.S. Dept. of Energy).  All rights reserved.
+#
+#
+#
+
+package Warewulf::Event::Nodelist;
+
+use Warewulf::Event;
+use Warewulf::EventHandler;
+use Warewulf::Logger;
+use Warewulf::RetVal;
+use Warewulf::DataStore;
+use Warewulf::Node;
+use File::Path;
+
+my $event = Warewulf::EventHandler->new();
+my $nodelist = "/etc/nodelist";
+```
+
+Like all Perl modules, the above is necessary for creating a package class, and including the libraries necessary for this code to function properly. Then we define our globals and construct the _$event_ object now (later it will allow us to assign subroutines to events).
+
+#### What we want to do
+
+```
+sub
+update_nodelist()
+{
+    my $datastore = Warewulf::DataStore->new();
+    my $nodeSet = $datastore->get_objects("node");
+
+    if (open(NODELIST, "> $nodelist")) {
+        foreach my $n ($nodeSet->get_list()) {
+            print NODELIST $n->name() ."\n";
```

Comment:
Hrmm... this should probably be chaned to `$n->nodename()` instead of `$n->name()`... but I'm not sure of original intention. @gmkurtzer ?